### PR TITLE
Silence warnings

### PR DESCRIPTION
**Pull request by atgeirr**
Fixing two sets of warnings from clang.

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
Do not merge this yet, it is incomplete still. More warnings...

**Comment by bska** [comment]
> Do not merge this yet, it is incomplete still. More warnings...

Noted.  Maybe you can reset the PR to a draft state to avoid inadvertent merging?

File: opm/material/fluidsystems/BlackOilFluidSystem.hpp [hunk 1]
```diff
@@ -1543,6 +1543,32 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
     static bool isInitialized_;
 };
 
+#define BOFS(T) BlackOilFluidSystem<T, BlackOilDefaultIndexTraits>
```
Comment (akva2) [inline opm/material/fluidsystems/BlackOilFluidSystem.hpp:1546]: template<class T> using BOFS = BlackoilFluidSystem<T, BlackOilDefaultIndexTraits>;

File: opm/material/fluidsystems/BlackOilFluidSystem.hpp [hunk 2]
```diff
@@ -1543,6 +1543,32 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
...
+template<> std::shared_ptr<OilPvtMultiplexer<T>> BOFS(T)::oilPvt_; \
+template<> std::shared_ptr<GasPvtMultiplexer<T>> BOFS(T)::gasPvt_; \
+template<> std::shared_ptr<WaterPvtMultiplexer<T>> BOFS(T)::waterPvt_; \
+template<> std::vector<std::array<T, 3>> BOFS(T)::referenceDensity_; \
+template<> std::vector<std::array<T, 3>> BOFS(T)::molarMass_; \
+template<> std::vector<std::array<T, 9>> BOFS(T)::diffusionCoefficients_; \
+template<> bool BOFS(T)::isInitialized_;
+
+//DECLARE_INSTANCE(float)
+DECLARE_INSTANCE(double)
```
Comment (akva2) [inline opm/material/fluidsystems/BlackOilFluidSystem.hpp:1570]: as this is a header pulled in all over the place please undef the macro when done.

**Comment by atgeirr** [comment]
Now it should be ok.

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
Made the requested changes @akva2.

**Review by akva2 — APPROVED** [review]

**Pull request merged**